shielder/contract/drink_tests/utils/shielder.rs

```diff
@@ -0,0 +1,141 @@
+use drink::{
+    runtime::MinimalRuntime,
+    session::{Session, NO_ARGS, NO_ENDOWMENT, NO_SALT},
+    AccountId32,
+};
+
+use crate::{
+    contract::OpPub,
+    drink_tests::BundleProvider,
+    mocked_zk::{
+        account::Account,
+        note::Note,
+        ops::{OpPriv, Operation},
+        relations::ZkProof,
+        tests::merkle::MerkleTree,
+        traits::Hashable,
+    },
+    types::Scalar,
+};
+
+pub struct ShielderUserEnv {
+    pub id: Scalar,
+    pub proof: ZkProof,
```

Comment:
what is this proof about? I mean, what statement does it prove?

Reply:
Basically, `ZkProof` is user's "knowledge" about id, trapdoor etc. `ShielderUserEnv` is convenient wrapper around ZkProof, extending "knowledge" to tree_leaf_id of ZkProof. For drink tests

Reply:
yeah, I kinda know what a proof is; but how does it relate to `id`, `nullifier` and `tree_leaf_id`? during doing next shielder action, like withdraw or deposit, you will have to prove more statements with other data... so what's the point of this object here?

Reply:
`nullifier` and `tree_leaf_id` are not contained in proof, so user has to take care of remembering/updating it
Of course, I can contain everything in the mocked proof and get rid of `ShielderUserEnv` struct, but I try to be consistent with what actual relations would look like.

removed `id` from struct, as it is a part of proof